easybuild/easyblocks/o/openssl_wrapper.py

```diff
@@ -0,0 +1,268 @@
+                    self.ssl_sysengines = engines_path
+                    self.log.debug("Found OpenSSL engines in: %s", self.ssl_sysengines)
+
+            if not self.ssl_sysengines:
+                self.ssl_syslib = None
+                print_warning("Found OpenSSL in host system, but not its engines."
+                              "Falling back to OpenSSL in EasyBuild")
+
+        # Check system include paths for OpenSSL headers
+        cmd = "gcc -E -Wp,-v -xc /dev/null"
+        (out, ec) = run_cmd(cmd, log_all=True, simple=False, trace=False)
+
+        sys_include_dirs = []
+        for match in re.finditer(r'^\s(/[^\0\n]*)+', out, re.MULTILINE):
+            sys_include_dirs.extend(match.groups())
+        self.log.debug("Found the following include directories in host system: %s", ', '.join(sys_include_dirs))
+
+        # headers are always in "include/openssl" subdirectories
+        ssl_include_dirs = [os.path.join(include, self.name.lower()) for include in sys_include_dirs]
+        ssl_include_dirs = [include for include in ssl_include_dirs if os.path.isdir(include)]
+
+        # verify that the headers match our OpenSSL version
+        for include_dir in ssl_include_dirs:
+            opensslv_path = os.path.join(include_dir, 'opensslv.h')
+            if os.path.exists(opensslv_path):
+                opensslv = read_file(opensslv_path)
+                header_majmin_version = re.search(r"SHLIB_VERSION_NUMBER\s\"([0-9]+\.[0-9]+)", opensslv, re.M)
+                if re.match("^{}".format(*header_majmin_version.groups()), self.version):
+                    self.ssl_sysheader = include_dir
+                    self.log.info("Found OpenSSL headers in host system: %s", self.ssl_sysheader)
+                    break
+
+        if not self.ssl_sysheader:
+            self.log.info("OpenSSL headers not found in host system")
+
+    def fetch_step(self, *args, **kwargs):
+        """Fetch sources if OpenSSL component is needed"""
+        if not all([self.ssl_syslib, self.ssl_sysheader]):
+            super(EB_OpenSSL_wrapper, self).fetch_step(*args, **kwargs)
+
+    def extract_step(self):
+        """Extract sources if OpenSSL component is needed"""
+        if not all([self.ssl_syslib, self.ssl_sysheader]):
+            super(EB_OpenSSL_wrapper, self).extract_step()
+
+    def install_step(self):
+        """Symlink target OpenSSL installation"""
+        shlib_ext = get_shared_lib_ext()
+
+        if self.ssl_syslib and self.ssl_sysheader:
+            # The host system already provides the necessary OpenSSL files
+            lib_pattern = ['%s*.%s*' % (lib_so, shlib_ext) for lib_so in self.openssl_libs]
+            lib_pattern = [os.path.join(os.path.dirname(self.ssl_syslib), '%s' % ptrn) for ptrn in lib_pattern]
+            lib_pattern.append(os.path.join(self.ssl_sysengines, '*'))
+
+            include_pattern = [os.path.join(self.ssl_sysheader, '*')]
+
+            bin_path = which(self.name.lower())
```

Comment:
```suggestion
            if self.version == '1.1':
                bin_path = which('openssl11', log_error=False)
                if bin_path is None:
                    bin_path = which(self.name.lower())
            else:
                bin_path = which(self.name.lower())
```

Note: the code doesn't check if this is the right version of `openssl` is found. Perhaps it's best to do this in the configure step asl well by introducing a `self.ssl_sysbin`, since we technically need that as well.

Reply:
@Micket I also improved the sanity check command to check the actual version of the binary in https://github.com/easybuilders/easybuild-easyblocks/pull/2429/commits/25d718c26bc846c5893cb505c0c491820423570d